src/mesh/Mesh.cpp

```diff
@@ -209,7 +209,10 @@ void Mesh::allocateDataValues()
 void Mesh::computeBoundingBox()
 {
   PRECICE_TRACE(_name);
-  BoundingBox bb(_dimensions);
+
+  // Keep the bounding box if set via the API function.
+  BoundingBox bb = _boundingBox.empty() ? BoundingBox(_dimensions) : BoundingBox(_boundingBox);
```

Comment:
If the user sets a BB, then this Mesh should be empty and expanding it is a noop, right?
In that case, you could simply exit the function `return;`

How does this impact tests and the partitioning where `computeBoundingBox()` may be called multiple times on the same mesh?
Recomputing the BB will then be interpreted as "user set it via API".

Reply:
> If the user sets a BB, then this Mesh should be empty and expanding it is a noop, right?

Not. setting BBs and using local meshes can be combined. Example: You want to have a read mapping from the received to your local mesh, but you want to directly write into the received mesh. And you need a larger part of the received mesh for writing than the read mapping needs.

Reply:
And how does it affect recomputing the boundingboxes?

Reply:
Do you have an example where we recompute bounding boxes?

Reply:
This is necessary for adaptive meshes aka my thesis.

Reply:
Then I would say it depends how you implement them and how they are supposed to work together with a direct access. I guess one of the following two options should work: delete the previous bounding boxes and 1) recompute them or 2) require the user to provide a new one. Do you agree?

Reply:
So, resetting the mesh will also reset the bounding box. I would say that this sounds like expected behaviour.

This means that I will document it once I enable `resetMesh()`.